### PR TITLE
refactor: optimize flags and SetLabel

### PR DESCRIPTION
**Pull request by dsseng**
Do not do string lookups in repetitive calls. We do not support changing SELinux status during runtime, so once we read this we can assume status does not change.

Also avoid unneeded FS writes when appropriate label is already set on file.

Signed-off-by: Dmitry Sharshakov <dmitry.sharshakov@siderolabs.com>

### CODE REVIEW

File: internal/pkg/selinux/selinux.go [hunk 1]
```diff
@@ -6,54 +6,84 @@
...
 	"github.com/siderolabs/go-procfs/procfs"
-	"golang.org/x/sys/unix"
 
 	"github.com/siderolabs/talos/pkg/machinery/constants"
 )
 
 //go:embed policy/policy.33
 var policy []byte
 
+var isEnabled, checkedEnabled, isEnforcing, checkedEnforcing bool
```
Comment (DmitriyMV) [inline internal/pkg/selinux/selinux.go:23]: I know this code is sequential, but I really don't like the idea of the global mutable variables. Also `is|checked` combo can be replaced with `optional.Optional[bool]`.

Comment (dsseng) [reply internal/pkg/selinux/selinux.go:23]: Another option could be defining functions as constants using [OnceValue](https://pkg.go.dev/sync#OnceValue). To me global variable also seems not a really great idea, so perhaps this is more atomic.

Comment (albert849) [reply internal/pkg/selinux/selinux.go:23]: Great job

Comment (smira) [reply internal/pkg/selinux/selinux.go:23]: I think `IsEnabled`/`IsEnforcing` is better as `sync.OnceValue` - we can't guarantee they won't be called concurrently.

Comment (DmitriyMV) [reply internal/pkg/selinux/selinux.go:23]: I agree. I think you can store `sync.OnceValue` inside unexported variable, move logic there and call this variable from the existing function.

Comment (dsseng) [reply internal/pkg/selinux/selinux.go:23]: Yes, will do like this

File: internal/pkg/selinux/selinux.go [hunk 2]
```diff
@@ -6,54 +6,75 @@
...
 	val := procfs.ProcCmdline().Get(constants.KernelParamSELinux).First()
 
 	return val != nil && *val == "1"
+})
+
+// IsEnabled checks if SELinux is enabled on the system by reading
+// the kernel command line. It returns true if SELinux is enabled,
+// otherwise it returns false. It also ensures we're not in a container.
+// By default SELinux is disabled.
+func IsEnabled() bool {
```
Comment (smira) [inline internal/pkg/selinux/selinux.go:38]: We could probably have var IsEnabled= sync.OnceValue

**Review by smira — APPROVED** [review]

**Pull request merged**